From 9d69fe8fe66b88c612a0fb6428687ea788465dde Mon Sep 17 00:00:00 2001
From: Ronan Abhamon <ronan.abhamon@vates.fr>
Date: Thu, 2 Jun 2022 09:28:32 +0200
Subject: [PATCH 063/171] fix(linstorvhdutil): fix boolean params of `check`
 call

`ignoreMissingFooter` and `fast` must be string types to be used with XAPI plugin API.

Signed-off-by: Ronan Abhamon <ronan.abhamon@vates.fr>
---
 drivers/linstor-manager   | 6 ++++--
 drivers/linstorvhdutil.py | 5 ++++-
 2 files changed, 8 insertions(+), 3 deletions(-)

diff --git a/drivers/linstor-manager b/drivers/linstor-manager
index c6d622f..63c0e3e 100755
--- a/drivers/linstor-manager
+++ b/drivers/linstor-manager
@@ -289,8 +289,10 @@ def destroy(session, args):
 def check(session, args):
     try:
         device_path = args['devicePath']
-        ignore_missing_footer = args['ignoreMissingFooter']
-        fast = args['fast']
+        ignore_missing_footer = distutils.util.strtobool(
+            args['ignoreMissingFooter']
+        )
+        fast = distutils.util.strtobool(args['fast'])
         return str(vhdutil.check(device_path, ignore_missing_footer, fast))
     except Exception as e:
         util.SMlog('linstor-manager:check error: {}'.format(e))
diff --git a/drivers/linstorvhdutil.py b/drivers/linstorvhdutil.py
index 9ba0ac3..f3d9870 100644
--- a/drivers/linstorvhdutil.py
+++ b/drivers/linstorvhdutil.py
@@ -111,7 +111,10 @@ class LinstorVhdUtil:
     # --------------------------------------------------------------------------
 
     def check(self, vdi_uuid, ignore_missing_footer=False, fast=False):
-        kwargs = {'ignoreMissingFooter': ignore_missing_footer, 'fast': fast}
+        kwargs = {
+            'ignoreMissingFooter': str(ignore_missing_footer),
+            'fast': str(fast)
+        }
         return self._check(vdi_uuid, **kwargs)
 
     @linstorhostcall(vhdutil.check, 'check')
-- 
2.44.0

